source/slang/slang-emit-c-like.cpp

```diff
@@ -884,14 +883,32 @@ void CLikeSourceEmitter::emitSimpleValueImpl(IRInst* inst)
                     // This little hack is needed for gcc that if we have the expression
                     // int(-0x80000000) we get the warning: warning :  integer overflow in expression [-Woverflow]
                     // 0x80000000 and -0x80000000 mean the same thing when casted to 32 bit int, so we just flip the value here.
-                    IRIntegerValue value = litInst->value.intVal;
-                    value = (value == -0x80000000ll) ? -value : value;
-
+                    IRIntegerValue value = (litInst->value.intVal & 0xFFFFFFFF);
                     m_writer->emit("int(");
-                    m_writer->emit(value);
+                    m_writer->emit(int32_t(value));
                     m_writer->emit(")");
                     return;
                 }
+                case BaseType::Int8:
```

Comment:
It seems from below that the construction of the integer literal handles masking/sign extension as part of construction. If that's the case doing the sign extension/masking here isn't necessary?

Reply:
Currently I am not 100% confident that the masking we done at construction covers 100% sources of constant insts, so adding the logic here is just to be safe.